### ISSUE TITLE
Imports in the conf/rexster-cassandra.xml and conf/rexster-cassandra-es.xml are ignored

### ISSUE THREAD

**Issue by angelcervera** [thinkaurelius/titan#869]
Test:

gremlin> xx = mgmt.makeEdgeLabel('xx').multiplicity(MANY2ONE).make()
==>javax.script.ScriptException: groovy.lang.MissingPropertyException: No such property: MANY2ONE for class: Script2

gremlin> xx = mgmt.makeEdgeLabel('xx').multiplicity(Multiplicity.MANY2ONE).make()
==>javax.script.ScriptException: groovy.lang.MissingPropertyException: No such property: Multiplicity for class: Script1

gremlin> xx = mgmt.makeEdgeLabel('xx').multiplicity(com.thinkaurelius.titan.core.Multiplicity.MANY2ONE).make()
==>xx

**Comment by dalaro** [comment]
What version is this?  On titan05 HEAD (0.5.3-SNAPSHOT), Multiplicity resolves but its enum values do not:

```
rexster[groovy]> Multiplicity
==>class com.thinkaurelius.titan.core.Multiplicity
rexster[groovy]> MANY2ONE
==>An error occurred while processing the script for language [groovy]. All transactions across all graphs in the session have been concluded with failure: java.util.concurrent.ExecutionException: javax.script.ScriptException: javax.script.ScriptException: groovy.lang.MissingPropertyException: No such property: MANY2ONE for class: Script2
rexster[groovy]>
```

```
gremlin> Multiplicity
==>class com.thinkaurelius.titan.core.Multiplicity
gremlin> MANY2ONE
No such property: MANY2ONE for class: groovysh_evaluate
Display stack trace? [yN]
```

The problem description doesn't completely make sense to me.  The import lists in in conf/rexster-cassandra.xml and conf/rexster-cassandra-es.xml apply only to Rexster.  `gremlin>` looks like the gremlin.sh prompt.  The conf files with rexster in their names have no bearing on the `gremlin>`/gremlin.sh environment.  However, this is a minor quibble, since the list in the Rexster files is generated from those used by gremlin.sh.  Furthermore, the docs include the example `mgmt.makeEdgeLabel('mother').multiplicity(MANY2ONE).make()`, but this is an error in the console:

```
gremlin> mgmt.makeEdgeLabel('mother').multiplicity(MANY2ONE).make()
No such property: MANY2ONE for class: groovysh_evaluate
Display stack trace? [yN] n
gremlin> mgmt.makeEdgeLabel('mother').multiplicity(Multiplicity.MANY2ONE).make()
==>mother
```

So at a minimum there's disagreement between the docs and the current import set w.r.t Multiplicity.  Cardinality has one instance of the same problem.

**Comment by dalaro** [comment]
On the one hand, the redundancy of `multiplicity(Multiplicity.MANY2ONE)` is silly.  On the other hand, importing identifiers as basic as `SET`, `LIST`, and `SINGLE` with special schema meanings is an invitation to unintended aliasing in application code that uses the same identifiers by coincidence.  This is a lesser-evil choice...

**Comment by angelcervera** [comment]
I am using the last release, 5.0.2 and **Rexster Dog House Gremlin Shell**.

**Comment by dalaro** [comment]
I made a couple of matter-of-taste changes:
- Multiplicity.\* and Cardinality.\* enum values (e.g. `SET` and `MANY2ONE`) will be imported by default in the gremlin.sh and rexster-console.sh shipped with Titan 0.5.3.  e91119524b87a2d87f4cf23066bbc0866634464e
- As I found them, the code examples in the docs used a mixture of unqualified enum values (e.g. MANY2ONE) and qualified ones (e.g. Multiplicity.MULTI).  Now all the Cardinality and Multiplicity uses in the doc code examples are qualified.  This tends toward repetition, as in `cardinality(Cardinality.SET)`, but I'm hoping that will be easier to understand for new users than the more concise but less-explicit `cardinality(SET)`.  Readers who understand what's going on and want the shorter form in their code can still static import the enum values they need (and they'll have those imports by default in 0.5.3's shell scripts).  9d38c2423baabcfa4ec8b26db9d898dfa405271d

@angelcervera I think something else is going on in your case. When I tested on titan05, `Multiplicity` didn't resolve but its values, such as `MANY2ONE`, did resolve.  Your paste shows that even `Multiplicity` doesn't resolve.  So I wasn't able to reproduce what you showed.  Did you download the titan-0.5.2-hadoop{1,2}.zip file, or did you download the rexster-server zipfile and install Titan as an extension?  The latter might explain what's going on here.  Please reopen if you can share more info.

**Comment by angelcervera** [comment]
@dalaro Sorry. I was fighting in other battles.

**Step to reproduce the error with version 0.5.2:**

**Server info:**
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.1 LTS
Release:    14.04
Codename:   trust

**Java Version:**
java version "1.8.0_20"
Java(TM) SE Runtime Environment (build 1.8.0_20-b26)
Java HotSpot(TM) 64-Bit Server VM (build 25.20-b23, mixed mode)
All clean, only JDK. Nothing added to the classpath.
- Download **titan-0.5.2-hadoop2.zip** and uncompress.
- Update base-uri tag in rexster-cassandra-es.xml and rexster-cassandra.xml
- start Titan with bin/titan.sh start
- Connect to rexter using the gremlin console in "The Dog House": **http://192.168.1.155:8182/doghouse/main/gremlin/graph**

**Execute:**
mgmt = g.getManagementSystem();
follow = mgmt.makeEdgeLabel('follow').multiplicity(MULTI).make();
Throws:
==>javax.script.ScriptException: groovy.lang.MissingPropertyException: No such property: MULTI for class: Script2

**Step to reproduce the error with version 0.5.3:**
- git clone https://github.com/thinkaurelius/titan.git titan05
- mvn package -Dmaven.test.skip=true

Result:
[INFO] Titan: Distributed Graph Database ................. SUCCESS [0.928s]
[INFO] Titan-Core: Core Library for Titan ................ SUCCESS [24.656s]
[INFO] Titan-Test: Test Suite for Titan .................. SUCCESS [3.717s]
[INFO] Titan-BerkeleyJE: Distributed Graph Database ...... SUCCESS [0.377s]
[INFO] Titan-Cassandra: Distributed Graph Database ....... SUCCESS [5.008s]
[INFO] Titan-ElasticSearch: Distributed Indexing Support . SUCCESS [8.922s]
[INFO] Titan-HBase: Parent Module ........................ SUCCESS [0.173s]
[INFO] Titan-HBase: Version-independent Core ............. SUCCESS [12.068s]
[INFO] Titan-HBase: 0.94 Compatibility Shim .............. FAILURE [2.271s]
[INFO] Titan-HBase: 0.96 Compatibility Shim .............. SKIPPED
[INFO] Titan-HBase: 0.98 Compatibility Shim .............. SKIPPED
[INFO] Titan-HBase: Universal binary ..................... SKIPPED
[INFO] Titan-Hadoop: Parent Module ....................... SKIPPED
[INFO] Titan-Hadoop: Version-independent Core ............ SKIPPED
[INFO] Titan-Hadoop: 1.x Compatibility Shim .............. SKIPPED
[INFO] Titan-Hadoop: 2.x Compatibility Shim .............. SKIPPED
[INFO] Titan-Hadoop: Universal binary .................... SKIPPED
[INFO] Titan-Lucene: Indexing Support .................... SKIPPED
[INFO] Titan-Rexster: Support for Titan in Rexster ....... SKIPPED
[INFO] Titan-Solr: Distributed Indexing Support .......... SKIPPED
[INFO] Titan-All: Complete Titan Distribution ............ SKIPPED
[INFO] Titan-Dist: Tar and Zip Archives .................. SKIPPED
[INFO] Titan-Dist: Archive with Hadoop 1 ................. SKIPPED
[INFO] Titan-Dist: Archive with Hadoop 2 ................. SKIPPED
[INFO] ------------------------------------------------------------------------
**[INFO] BUILD FAILURE**
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 59.235s
[INFO] Finished at: Fri Dec 19 07:53:50 GMT 2014
[INFO] Final Memory: 67M/239M
[INFO] ------------------------------------------------------------------------
**[ERROR] Failed to execute goal on project titan-hbase-094: Could not resolve dependencies for project com.thinkaurelius.titan:titan-hbase-094:jar:0.5.3-SNAPSHOT: Failure to find com.thinkaurelius.titan:titan-hbase-core:jar:tests:0.5.3-SNAPSHOT in https://oss.sonatype.org/content/repositories/snapshots was cached in the local repository, resolution will not be reattempted until the update interval of sonatype-nexus-snapshots has elapsed or updates are forced -> [Help 1]**

**So no lucky, it is not possible a quick test with the trunk version.**

Regards.

**Comment by dalaro** [comment]
@angelcervera Thanks for following up.

Replace `-Dmaven.test.skip=true` with `-DskipTests`.  The former prevents the test classes from compiling and packaging and ultimately leads to dependency resolution failures like the one you pasted.  The latter builds and packages the tests, allowing test deps to resolve (it just doesn't run them).

Your last reply shows this problem:

```
follow = mgmt.makeEdgeLabel('follow').multiplicity(MULTI).make();
Throws:
==>javax.script.ScriptException: groovy.lang.MissingPropertyException: No such property: MULTI for class: Script2
```

I think that's already understood and fixed on titan05.  I was able to reproduce that one on my end using titan-0.5.2-hadoop2.zip.

I'm much more interested in this similar but distinct failure to resolve `Multiplicity` pasted in your original report, which I couldn't reproduce:

```
gremlin> xx = mgmt.makeEdgeLabel('xx').multiplicity(Multiplicity.MANY2ONE).make()
==>javax.script.ScriptException: groovy.lang.MissingPropertyException: No such property: Multiplicity for class: Script1
```

**Comment by angelcervera** [comment]
@dalaro Thanks you for your help.

Also errors with -DskipTests and checking that it is not resolved Multiplicity on 0.5.2

[INFO] Titan: Distributed Graph Database ................. SUCCESS [0.258s]
[INFO] Titan-Core: Core Library for Titan ................ SUCCESS [22.639s]
[INFO] Titan-Test: Test Suite for Titan .................. SUCCESS [0.311s]
[INFO] Titan-BerkeleyJE: Distributed Graph Database ...... SUCCESS [0.213s]
[INFO] Titan-Cassandra: Distributed Graph Database ....... SUCCESS [0.545s]
[INFO] Titan-ElasticSearch: Distributed Indexing Support . SUCCESS [0.888s]
[INFO] Titan-Lucene: Indexing Support .................... SUCCESS [0.121s]
[INFO] Titan-Rexster: Support for Titan in Rexster ....... SUCCESS [0.065s]
[INFO] Titan-Solr: Distributed Indexing Support .......... SUCCESS [0.822s]
[INFO] Titan-All: Complete Titan Distribution ............ SUCCESS [0.865s]
[INFO] Titan-Dist: Tar and Zip Archives .................. FAILURE [0.108s]
[INFO] Titan-Dist: Archive with Hadoop 1 ................. SKIPPED
[INFO] Titan-Dist: Archive with Hadoop 2 ................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 27.180s
[INFO] Finished at: Fri Dec 19 20:15:51 GMT 2014
[INFO] Final Memory: 43M/289M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project titan-dist: Could not resolve dependencies for project com.thinkaurelius.titan:titan-dist:pom:0.5.3-SNAPSHOT: Failure to find com.thinkaurelius.titan:titan-hbase-core:jar:tests:0.5.3-SNAPSHOT in https://oss.sonatype.org/content/repositories/snapshots was cached in the local repository, resolution will not be reattempted until the update interval of sonatype-nexus-snapshots has elapsed or updates are forced -> [Help 1]
[ERROR] 

**Trying to resolve Multiplicity class**

> gremlin> mgmt = g.getManagementSystem();
> gremlin> xx = mgmt.makeEdgeLabel('xx').multiplicity(Multiplicity.MANY2ONE).make();
> ==>javax.script.ScriptException: groovy.lang.MissingPropertyException: No such property: > Multiplicity for class: Script2
> gremlin> Multiplicity
> ==>javax.script.ScriptException: groovy.lang.MissingPropertyException: No such property: Multiplicity for class: Script3

If I have time this weekend, I will try to compile the version 0.5.3-SNAPSHOT

Regards

**Comment by dalaro** [comment]
Thanks for following up.

I ran these commands on titan05 and again on 0.5.2:

```
rm -rf ~/.m2/repository/com/thinkaurelius/
mvn clean install -DskipTests=true
```

and the build succeeded both times.  Do _not_ run `mvn clean install -DskipTests=true -Dmaven.test.skip=true`.

Almost everything about the command output in your last post seems strange to me.  For example: why does Maven's list of modules in the build reactor omit the hbase modules and the hadoop modules (have the poms been modified?).  Another example: if the build failed, then how did you execute that gremlin snippet?

**Comment by angelcervera** [comment]
Hi @dalaro 

> Another example: if the build failed, then how did you execute that gremlin snippet? 

I said "checking that it is not resolved Multiplicity on 0.5.2" and "If I have time this weekend, I will try to compile the version 0.5.3-SNAPSHOT" So _my test was with version 0.5.2_
I explained all step to reproduce the error in previous post.

> For example: why does Maven's list of modules in the build reactor omit the hbase modules and the hadoop modules (have the poms been modified?)

No, I did not modify anything. Only checkout and install. And only with -DskipTests=true. The list of commands to reproduce:

> git clone https://github.com/thinkaurelius/titan.git titan05
> cd titan05
> rm -rf ~/.m2/repository/com/thinkaurelius/
> mvn clean install -DskipTests=true
> 
> [INFO] ------------------------------------------------------------------------
> [INFO] Reactor Summary:
> [INFO] 
> [INFO] Titan: Distributed Graph Database ................. SUCCESS [0.333s]
> [INFO] Titan-Core: Core Library for Titan ................ SUCCESS [19.371s]
> [INFO] Titan-Test: Test Suite for Titan .................. SUCCESS [1.964s]
> [INFO] Titan-BerkeleyJE: Distributed Graph Database ...... SUCCESS [0.239s]
> [INFO] Titan-Cassandra: Distributed Graph Database ....... SUCCESS [1.039s]
> [INFO] Titan-ElasticSearch: Distributed Indexing Support . SUCCESS [0.753s]
> [INFO] Titan-HBase: Parent Module ........................ SUCCESS [0.282s]
> [INFO] Titan-HBase: Version-independent Core ............. SUCCESS [1.080s]
> [INFO] Titan-HBase: 0.94 Compatibility Shim .............. SUCCESS [3.279s]
> [INFO] Titan-HBase: 0.96 Compatibility Shim .............. SUCCESS [2.521s]
> [INFO] Titan-HBase: 0.98 Compatibility Shim .............. FAILURE [1.156s]
> [INFO] Titan-HBase: Universal binary ..................... SKIPPED
> [INFO] Titan-Hadoop: Parent Module ....................... SKIPPED
> [INFO] Titan-Hadoop: Version-independent Core ............ SKIPPED
> [INFO] Titan-Hadoop: 1.x Compatibility Shim .............. SKIPPED
> [INFO] Titan-Hadoop: 2.x Compatibility Shim .............. SKIPPED
> [INFO] Titan-Hadoop: Universal binary .................... SKIPPED
> [INFO] Titan-Lucene: Indexing Support .................... SKIPPED
> [INFO] Titan-Rexster: Support for Titan in Rexster ....... SKIPPED
> [INFO] Titan-Solr: Distributed Indexing Support .......... SKIPPED
> [INFO] Titan-All: Complete Titan Distribution ............ SKIPPED
> [INFO] Titan-Dist: Tar and Zip Archives .................. SKIPPED
> [INFO] Titan-Dist: Archive with Hadoop 1 ................. SKIPPED
> [INFO] Titan-Dist: Archive with Hadoop 2 ................. SKIPPED
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD FAILURE
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time: 32.465s
> [INFO] Finished at: Tue Dec 23 21:46:49 GMT 2014
> [INFO] Final Memory: 81M/313M
> [INFO] ------------------------------------------------------------------------
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:1.3.1:enforce (enforce-dependency-convergence) on project titan-hbase-098: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed. -> [Help 1]
> [ERROR] 
> [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
> [ERROR] Re-run Maven using the -X switch to enable full debug logging.
> [ERROR] 
> [ERROR] For more information about the errors and possible solutions, please read the following articles:
> [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
> [ERROR] 
> [ERROR] After correcting the problems, you can resume the build with the command
> [ERROR]   mvn <goals> -rf :titan-hbase-098

My version of maven:

> mvn --version
> Apache Maven 3.1.1 (0728685237757ffbf44136acec0402957f723d9a; 2013-09-17 16:22:22+0100)
> Maven home: /home/angel/applications/java/maven
> Java version: 1.8.0_11, vendor: Oracle Corporation
> Java home: /usr/local/java/jdk1.8.0_11/jre
> Default locale: es_ES, platform encoding: UTF-8
> OS name: "linux", version: "3.13.0-43-generic", arch: "amd64", family: "unix"

And this is the reactor calculated by maven:

> [INFO] Scanning for projects...
> [INFO] ------------------------------------------------------------------------
> [INFO] Reactor Build Order:
> [INFO] 
> [INFO] Titan: Distributed Graph Database
> [INFO] Titan-Core: Core Library for Titan
> [INFO] Titan-Test: Test Suite for Titan
> [INFO] Titan-BerkeleyJE: Distributed Graph Database
> [INFO] Titan-Cassandra: Distributed Graph Database
> [INFO] Titan-ElasticSearch: Distributed Indexing Support
> [INFO] Titan-HBase: Parent Module
> [INFO] Titan-HBase: Version-independent Core
> [INFO] Titan-HBase: 0.94 Compatibility Shim
> [INFO] Titan-HBase: 0.96 Compatibility Shim
> [INFO] Titan-HBase: 0.98 Compatibility Shim
> [INFO] Titan-HBase: Universal binary
> [INFO] Titan-Hadoop: Parent Module
> [INFO] Titan-Hadoop: Version-independent Core
> [INFO] Titan-Hadoop: 1.x Compatibility Shim
> [INFO] Titan-Hadoop: 2.x Compatibility Shim
> [INFO] Titan-Hadoop: Universal binary
> [INFO] Titan-Lucene: Indexing Support
> [INFO] Titan-Rexster: Support for Titan in Rexster
> [INFO] Titan-Solr: Distributed Indexing Support
> [INFO] Titan-All: Complete Titan Distribution
> [INFO] Titan-Dist: Tar and Zip Archives
> [INFO] Titan-Dist: Archive with Hadoop 1
> [INFO] Titan-Dist: Archive with Hadoop 2

More error details:

> [WARNING] 
> Dependency convergence error for jdk.tools:jdk.tools:1.7 paths to dependency are:
> +-com.thinkaurelius.titan:titan-hbase-098:0.5.3-SNAPSHOT
>   +-org.apache.hbase:hbase-client:0.98.8-hadoop2
>     +-org.apache.hbase:hbase-annotations:0.98.8-hadoop2
>       +-jdk.tools:jdk.tools:1.7
> and
> +-com.thinkaurelius.titan:titan-hbase-098:0.5.3-SNAPSHOT
>   +-org.apache.hadoop:hadoop-annotations:2.2.0
>     +-jdk.tools:jdk.tools:1.6
> 
> [DEBUG] Adding failure due to exception
> org.apache.maven.enforcer.rule.api.EnforcerRuleException: Failed while enforcing releasability the error(s) are [
> Dependency convergence error for jdk.tools:jdk.tools:1.7 paths to dependency are:
> +-com.thinkaurelius.titan:titan-hbase-098:0.5.3-SNAPSHOT
>   +-org.apache.hbase:hbase-client:0.98.8-hadoop2
>     +-org.apache.hbase:hbase-annotations:0.98.8-hadoop2
>       +-jdk.tools:jdk.tools:1.7
> and
> +-com.thinkaurelius.titan:titan-hbase-098:0.5.3-SNAPSHOT
>   +-org.apache.hadoop:hadoop-annotations:2.2.0
>     +-jdk.tools:jdk.tools:1.6
> ]
>     at org.apache.maven.plugins.enforcer.DependencyConvergence.execute(DependencyConvergence.java:140)
>     at org.apache.maven.plugins.enforcer.EnforceMojo.execute(EnforceMojo.java:177)
>     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:106)
>     at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
>     at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
>     at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
>     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
>     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
>     at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
>     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
>     at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:317)
>     at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:152)
>     at org.apache.maven.cli.MavenCli.execute(MavenCli.java:555)
>     at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:214)
>     at org.apache.maven.cli.MavenCli.main(MavenCli.java:158)
>     at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>     at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>     at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>     at java.lang.reflect.Method.invoke(Method.java:483)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
> Caused by: org.apache.maven.enforcer.rule.api.EnforcerRuleException: Failed while enforcing releasability the error(s) are [
> Dependency convergence error for jdk.tools:jdk.tools:1.7 paths to dependency are:
> +-com.thinkaurelius.titan:titan-hbase-098:0.5.3-SNAPSHOT
>   +-org.apache.hbase:hbase-client:0.98.8-hadoop2
>     +-org.apache.hbase:hbase-annotations:0.98.8-hadoop2
>       +-jdk.tools:jdk.tools:1.7
> and
> +-com.thinkaurelius.titan:titan-hbase-098:0.5.3-SNAPSHOT
>   +-org.apache.hadoop:hadoop-annotations:2.2.0
>     +-jdk.tools:jdk.tools:1.6
> ]
>     at org.apache.maven.plugins.enforcer.DependencyConvergence.execute(DependencyConvergence.java:131)
>     ... 22 more
> [WARNING] Rule 0: org.apache.maven.plugins.enforcer.DependencyConvergence failed with message:
> Failed while enforcing releasability the error(s) are [
> Dependency convergence error for jdk.tools:jdk.tools:1.7 paths to dependency are:
> +-com.thinkaurelius.titan:titan-hbase-098:0.5.3-SNAPSHOT
>   +-org.apache.hbase:hbase-client:0.98.8-hadoop2
>     +-org.apache.hbase:hbase-annotations:0.98.8-hadoop2
>       +-jdk.tools:jdk.tools:1.7
> and
> +-com.thinkaurelius.titan:titan-hbase-098:0.5.3-SNAPSHOT
>   +-org.apache.hadoop:hadoop-annotations:2.2.0
>     +-jdk.tools:jdk.tools:1.6
> ]
> [INFO] ------------------------------------------------------------------------
> [INFO] Reactor Summary:
> [INFO] 
> [INFO] Titan: Distributed Graph Database ................. SUCCESS [0.303s]
> [INFO] Titan-Core: Core Library for Titan ................ SUCCESS [16.160s]
> [INFO] Titan-Test: Test Suite for Titan .................. SUCCESS [2.492s]
> [INFO] Titan-BerkeleyJE: Distributed Graph Database ...... SUCCESS [0.347s]
> [INFO] Titan-Cassandra: Distributed Graph Database ....... SUCCESS [1.255s]
> [INFO] Titan-ElasticSearch: Distributed Indexing Support . SUCCESS [1.180s]
> [INFO] Titan-HBase: Parent Module ........................ SUCCESS [0.343s]
> [INFO] Titan-HBase: Version-independent Core ............. SUCCESS [1.774s]
> [INFO] Titan-HBase: 0.94 Compatibility Shim .............. SUCCESS [2.420s]
> [INFO] Titan-HBase: 0.96 Compatibility Shim .............. SUCCESS [1.617s]
> [INFO] Titan-HBase: 0.98 Compatibility Shim .............. FAILURE [1.159s]
> [INFO] Titan-HBase: Universal binary ..................... SKIPPED
> [INFO] Titan-Hadoop: Parent Module ....................... SKIPPED
> [INFO] Titan-Hadoop: Version-independent Core ............ SKIPPED
> [INFO] Titan-Hadoop: 1.x Compatibility Shim .............. SKIPPED
> [INFO] Titan-Hadoop: 2.x Compatibility Shim .............. SKIPPED
> [INFO] Titan-Hadoop: Universal binary .................... SKIPPED
> [INFO] Titan-Lucene: Indexing Support .................... SKIPPED
> [INFO] Titan-Rexster: Support for Titan in Rexster ....... SKIPPED
> [INFO] Titan-Solr: Distributed Indexing Support .......... SKIPPED
> [INFO] Titan-All: Complete Titan Distribution ............ SKIPPED
> [INFO] Titan-Dist: Tar and Zip Archives .................. SKIPPED
> [INFO] Titan-Dist: Archive with Hadoop 1 ................. SKIPPED
> [INFO] Titan-Dist: Archive with Hadoop 2 ................. SKIPPED
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD FAILURE
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time: 29.556s
> [INFO] Finished at: Tue Dec 23 22:20:31 GMT 2014
> [INFO] Final Memory: 80M/304M
> [INFO] ------------------------------------------------------------------------
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:1.3.1:enforce (enforce-dependency-convergence) on project titan-hbase-098: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed. -> [Help 1]
> org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:1.3.1:enforce (enforce-dependency-convergence) on project titan-hbase-098: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed.
>     at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:216)
>     at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
>     at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
>     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
>     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
>     at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
>     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
>     at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:317)
>     at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:152)
>     at org.apache.maven.cli.MavenCli.execute(MavenCli.java:555)
>     at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:214)
>     at org.apache.maven.cli.MavenCli.main(MavenCli.java:158)
>     at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>     at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>     at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>     at java.lang.reflect.Method.invoke(Method.java:483)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
>     at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
> Caused by: org.apache.maven.plugin.MojoExecutionException: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed.
>     at org.apache.maven.plugins.enforcer.EnforceMojo.execute(EnforceMojo.java:209)
>     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:106)
>     at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
>     ... 19 more
> [ERROR] 
> [ERROR] 
> [ERROR] For more information about the errors and possible solutions, please read the following articles:
> [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
> [ERROR] 
> [ERROR] After correcting the problems, you can resume the build with the command
> [ERROR]   mvn <goals> -rf :titan-hbase-098

Maybe you need test in a clean machine. Are you using Jenkins or Hudson or other CI? Maybe will be usesful.

**Comment by dalaro** [comment]
This error:

```
+-com.thinkaurelius.titan:titan-hbase-098:0.5.2
  +-org.apache.hbase:hbase-client:0.98.8-hadoop2
    +-org.apache.hbase:hbase-annotations:0.98.8-hadoop2
      +-jdk.tools:jdk.tools:1.7
and
+-com.thinkaurelius.titan:titan-hbase-098:0.5.2
  +-org.apache.hadoop:hadoop-annotations:2.2.0
    +-jdk.tools:jdk.tools:1.6

[WARNING] Rule 0: org.apache.maven.plugins.enforcer.DependencyConvergence failed with message:
Failed while enforcing releasability the error(s) are [
Dependency convergence error for jdk.tools:jdk.tools:1.7 paths to dependency are:
+-com.thinkaurelius.titan:titan-hbase-098:0.5.2
  +-org.apache.hbase:hbase-client:0.98.8-hadoop2
    +-org.apache.hbase:hbase-annotations:0.98.8-hadoop2
      +-jdk.tools:jdk.tools:1.7
and
+-com.thinkaurelius.titan:titan-hbase-098:0.5.2
  +-org.apache.hadoop:hadoop-annotations:2.2.0
    +-jdk.tools:jdk.tools:1.6
```

occurs when building titan05 with JDK 8.  titan09 builds with JDK8, but titan05 only builds cleanly with JDK7 at the moment.

This build error is important -- it would be useful for titan05 to compile cleanly to a 1.7 target using JDK8 -- but I think it is orthogonal to the import problems at the root of this particular issue.  I also think those import issues are going to be resolved in the Titan zipfiles and docs shipped with 0.5.3.  More testing is always welcome, but I don't want to waste your time chasing build issues, so I'm closing this for now.   Please feel free to reopen if you want to pursue testing on this further.
